addons/mail/models/mail_message.py

```diff
@@ -150,6 +150,7 @@ def default_get(self, fields):
     starred = fields.Boolean(
         'Starred', compute='_compute_starred', search='_search_starred', compute_sudo=False,
         help='Current user has a starred notification linked to this message')
+    pinned_on_channel_id = fields.Many2one('mail.channel', 'pinned_message_ids')
```

Comment:
Ok to store on `mail.channel`, but the inverse `mail.message` should be `store=False`, so that we don't bloat this table more than it should.